server/webserver.cc

```diff
@@ -134,5 +134,6 @@ bool WebServer::run_server() {
     printf("Exception %s\n", e.what());
     return false;
   }
+  status_.GetStatus().destroyMutex();
```

Comment:
You can check if a object is valid (i.e. it still exists) before destructing it.
Either way the struct should never be copied by value